element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpConnectorTest.java

```diff
@@ -96,25 +91,24 @@ public void serverClientPingPong() throws Exception {
 		server.start();
 		client.start();
 
-		RawData msg = createMessage(new InetSocketAddress(port));
+		RawData msg = createMessage(server.getAddress(), messageSize, null, null);
 
 		client.send(msg);
 		serverCatcher.blockUntilSize(1);
 		assertArrayEquals(msg.getBytes(), serverCatcher.getMessage(0).getBytes());
 
 		// Response message must go over the same connection client already
 		// opened
-		msg = createMessage(serverCatcher.getMessage(0).getInetSocketAddress());
+		msg = createMessage(serverCatcher.getMessage(0).getInetSocketAddress(), messageSize, null, null);
 		server.send(msg);
 		clientCatcher.blockUntilSize(1);
 		assertArrayEquals(msg.getBytes(), clientCatcher.getMessage(0).getBytes());
 	}
 
 	@Test
 	public void singleServerManyClients() throws Exception {
-		int port = findEphemeralPort();
 		int clients = 100;
-		TcpServerConnector server = new TcpServerConnector(new InetSocketAddress(port), NUMBER_OF_THREADS,
+		TcpServerConnector server = new TcpServerConnector(new InetSocketAddress(0), NUMBER_OF_THREADS,
```

Comment:
shouldn't this be `new InetSocketAddress(InetAddress.getLoopbackAddress(), 0)` now?

I think there are several more occurrences ...